mozilla/foundation.mozilla.org
# ensure product/category pairs never have NULL categories

Closes #7814
Reverts #7776

This change updates the `ProductPageCategory` model (which represents a `(product <-> category)` many-to-many relation) so that if a category is deleted, any ProductPageCategory that uses that category is deleted, too, with a migration that explicitly deletes all currently extant ProductPageCategory instances with a category set to `None` (there are 171 entries in the production database for this at the time of writing this PR).

## Verifying the problem

- check out main
- copy the production database using `inv copy-prod-db`
- `docker-compose up`
- try to load http://localhost:8000/en/privacynotincluded/roku-streaming-sticks/
  - this should work fine
- log into the CMS
- go to the "Buyers Guide Product Categories" snippets section
- delete the "entertainment" category
- try to load http://localhost:8000/en/privacynotincluded/roku-streaming-sticks/
  - this will be a server error 500
- in the CMS, search for "roku streaming sticks" and edit the (en) version
  - the categories should show: ![image](https://user-images.githubusercontent.com/177243/140991773-5db994b6-c3b5-46ba-adb4-93c0f22f2652.png)
- delete the "empty" category from that list
- publish the product page
- try to load http://localhost:8000/en/privacynotincluded/roku-streaming-sticks/
  - the page should load fine again.
- try to load http://localhost:8000/en/privacynotincluded/nintendo-switch/
  - this should still be a server error 500. This will be fixed in the solution testing
- go to the edit page for the (en) nintendo switch
  - observe the "emtpy" category, but do not change/publish anything.

## Testing the solution

- `docker-compose down`
- check out this branch
- run `inv migrate`
- `docker-compose up`
- load http://localhost:8000/en/privacynotincluded/nintendo-switch/
  -  this should not have any errors
- go to the edit page for the (en) nintendo switch
  - there should not be an "empty" category in the edit view.
